resolwe_bio/management/commands/generate_etc.py

```diff
@@ -0,0 +1,128 @@
+        users = User.objects.filter(is_superuser=True).order_by('date_joined')
+        if not users.exists():
+            self.stderr.write("Admin does not exist: create a superuser")
+            exit(1)
+        return users.first()
+
+
+    @staticmethod
+    def create_etc(gene_ids, path, file_name):
+        genes = {}
+        with open(os.path.join(path, file_name), 'w') as f:
+            f.write('\t'.join(['Gene', '0', '4', '8', '12', '16', '20', '24']) + '\n')
+            with gzopen(gene_ids) as gene_ids:
+                all_genes = [line.strip() for line in gene_ids]
+                for gene in all_genes:
+                    expression = [str(random.gammavariate(1, 100)) for i in range(7)]
+                    expression = '\t'.join(expression)
+                    f.write('{}\t{}\n'.format(gene, expression))
+
+        with open(os.path.join(path, file_name)) as csvfile:
+            my_reader = csv.reader(csvfile, delimiter=str('\t'))
+            times = [int(x) for x in next(my_reader)[1:]]
+            for x in my_reader:
+                genes[escape_mongokey(str(x[0]))] = x[1:]
+
+        with open(os.path.join(path, 'etc.json'), 'w') as json_file:
+            js_out = '{{"etc":{}}}'.format(json.dumps({'genes': genes, 'timePoints': times}, separators=(',', ':')))
+            json_file.write(js_out)
+            gzip.open(os.path.join(path, 'etc.json.gz'), 'wb').write(js_out.encode('utf-8'))
+
+
+    @staticmethod
+    def generate_etc_desciptor():
+        project = [('1.', 'D. discoideum vs. D. purpureum'),
+                   ('2.', 'Filter Development vs. cAMP Pulsing; Frequent Sampling'),
+                   ('3.', 'GtaC: WT vs. mutants'),
+                   ('4.', 'lncRNA transcriptome')]
+
+        pn, pr = random.choice(project)
+
+        annotation = {'projectNumber': pn,
+                      'project': pr,
+                      'citation': {'name': 'Rosengarten et. al.',
+                                   'url': 'http://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1491-7'},
+                      'treatment': random.choice(['cAMP Pulses', 'Filter Development']),
+                      'parental_strain': 'AX4',
+                      'growth': 'K. pneumoniae'}
+
+        return annotation
+
+
+    def create_data(self, reads_name='seq_reads', rseed=None):
+        # get test data paths
+        data_dir = settings.FLOW_EXECUTOR['DATA_DIR']
+        test_files_path = os.path.abspath(
+            os.path.join(os.path.dirname(__file__), '..', '..', 'tests', 'processes', 'files'))
+        dicty_genes = os.path.join(test_files_path, 'dicty_genes.tab.gz')
+
+        # Create reads data object
+        started = timezone.now()
+        etc = Data.objects.create(
+            slug='etc',
+            name="D. discoideum",
+            started=started,
+            finished=started + datetime.timedelta(minutes=1),
+            descriptor_schema=DescriptorSchema.objects.get(slug='dicty-etc'),
```

Comment:
Check if DescriptorSchema exists before assigning it and raise meaningful if not.